iohelper/iohelper.go

```diff
@@ -0,0 +1,95 @@
+package iohelper
+
+/*
+ * This file contains structs and functions related to interacting with files
+ * and directories.
+ */
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/greenplum-db/gp-common-go-libs/gplog"
+	"github.com/greenplum-db/gp-common-go-libs/operating"
+	"github.com/pkg/errors"
+)
+
+/*
+ * Generic file/directory manipulation functions that use the function pointers
+ * and types from the operating package, for easier test mocking around file IO.
+ */
+
+func OpenFileForReading(filename string) (operating.ReadCloserAt, error) {
+	fileHandle, err := operating.System.OpenFileRead(filename, os.O_RDONLY, 0644)
+	if err != nil {
+		return nil, errors.Errorf("Unable to open file for reading: %s", err)
+	}
+	return fileHandle, nil
+}
+
+func MustOpenFileForReading(filename string) operating.ReadCloserAt {
+	fileHandle, err := OpenFileForReading(filename)
+	gplog.FatalOnError(err)
+	return fileHandle
+}
+
+func OpenFileForWriting(filename string, allowAppend ...bool) (io.WriteCloser, error) {
+	flags := os.O_CREATE | os.O_WRONLY
```

Comment:
I would rather see these options where we call the open file function rather than having them hidden in a wrapper